Shared/Objects/ItemLetterFilter.swift

```diff
@@ -0,0 +1,43 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2023 Jellyfin & Jellyfin Contributors
+//
+
+import Foundation
+import JellyfinAPI
+
+public enum ItemLetterFilter: Character, Codable, CaseIterable {
```

Comment:
I took some time to figure out the alphabet localization because I didn't like defining the alphabet manually like this.

The following API returns the locale alphabet + English alphabet + `#`, so we get this for free: https://developer.apple.com/documentation/uikit/uilocalizedindexedcollation

For example, in a Korean locale app the following would be returned:

```swift
print(UILocalizedIndexedCollation.current())

// ["ㄱ", "ㄴ", "ㄷ", "ㄹ", "ㅁ", "ㅂ", "ㅅ", "ㅇ", "ㅈ", "ㅊ", "ㅋ", "ㅌ", "ㅍ", "ㅎ", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "#"]
```

I know that I had previously mentioned that "we can do this later", but I would now like to do the localization since I would find it a bit less work than manually defining alphabets. As a part of this, I would like to change the contexts/names of everything to match Apple's, which I find more preferable and generic. For example, we would change the context of "letter" to "index" and even "alpha picker" to "index title selector". These aren't hard names so you may think of something better.

The `#` would be a new feature and that works with the `NameLessThan=A` parameter.

I anticipate that this would require refactoring and I would like consistent naming across the new feature both in code and that is displayed (`nameStartsWith` key vs `Letter`).

Reply:
Sounds good! I originally had # included but it gets a tiny bit messier with the logic. I can re-add this back in with the localization sometime this week!

Reply:
This is done. I am handling that switch over from 'nameStartsWith: #' -> 'nameLessThan: A' on the libraryModelView. Let me know if that isn't where that belongs.

Since I am using the full list of characters to populate the Filter the HStack Pill and the AlphaPicker, I made an email that just returns all characters sorted. I put my phone in Korean and this does work. 

The only issue/compromise that I made was that, with English+Other Language, this will likely exceed the static AlphaPicker that fit on all screens. My compromise was to turn the AlphaPicker into a ScrollView to better fit additional Characters. This isn't the end of the world to have it this way, especially since your currently scroll on the ScrollView isn't reset anymore because of your recommendations here: https://github.com/jellyfin/Swiftfin/pull/852#discussion_r1336236717

Let me know if the ScrollView is too detrimental (I personally think the static AlphaPicker is ideal). The other alternative is to only take the localization letters into account. That is, Korean localization would ONLY have Korean letters and no Romantic Letters. That's obtainable by using .IsLetter() since non-Romantic characters return false for that test. End of the day, the ScrollView isn't a bad solution but I don't love the move part right next to the libraryView being another scrolling, moving part.